File: backend-java/src/main/java/com/itxiang/evidence/common/Result.java
package com.itxiang.evidence.common;

import lombok.Data;

import java.io.Serializable;

@Data
public class Result<T> implements Serializable {

    private Integer code;
    private String msg;
    private T data;

    /**
     * 返回无数据的成功响应。
     */
    public static <T> Result<T> success() {
        Result<T> result = new Result<>();
        result.code = 1;
        return result;
    }

    /**
     * 返回携带业务数据的成功响应。
     */
    public static <T> Result<T> success(T object) {
        Result<T> result = new Result<>();
        result.data = object;
        result.code = 1;
        return result;
    }

    /**
     * 返回业务失败响应。
     */
    public static <T> Result<T> error(String msg) {
        Result<T> result = new Result<>();
        result.msg = msg;
        result.code = 0;
        return result;
    }
}

File: ai-python/app/core/result.py
"""对外业务接口的统一响应信封与可预期业务错误。"""

from __future__ import annotations

from typing import Generic, TypeVar

from pydantic import BaseModel


T = TypeVar("T")


class Result(BaseModel, Generic[T]):
    """保持与原 Java `Result<T>` 一致的响应结构。"""

    code: int
    msg: str | None = None
    data: T | None = None

    @classmethod
    def success(cls, data: T | None = None) -> "Result[T]":
        """构造成功响应。"""
        return cls(code=1, data=data)

    @classmethod
    def failure(cls, message: str) -> "Result[None]":
        """构造可安全展示给调用方的业务失败响应。"""
        return cls(code=0, msg=message, data=None)


class BusinessError(Exception):
    """表示应转换为 `Result.code=0` 的受控业务错误。"""

    def __init__(self, message: str) -> None:
        super().__init__(message)
        self.message = message
